### DISCUSSION TITLE
Beginner to this project

### DISCUSSION THREAD

**Discussion by himanshu1221** [Azure/azure-dev#2868] [Q&A] [answered]
1) How can we set this codebase in our machine I am new to it
2) I am still learning golang is there any prerequisite to contribute to this repository
3) Can anyone help me to grab some beginner-friendly issue as I am very new to it

**Comment by lechnerc77** [comment, marked as answer]
In general to contribute you need to work on your own fork of the repository. 

Ad 1) The repository contains a devcontainer definition. The container already has everything you need to develop and debug `azd`
So you can either use GitHub Codespaces directly on GitHub or you work locally and spin up a devcontainer.
Ad 2) the details about contributions are listed [here](https://github.com/Azure/azure-dev?tab=readme-ov-file#contributing) and in the [CONTRIBUTING.MD](https://github.com/Azure/azure-dev/blob/main/cli/azd/CONTRIBUTING.md).
Dependning on waht you want to contribute Golang is a prerequiste.
Ad 3) There are several labels that might help you:
  - ["good first issue"](https://github.com/Azure/azure-dev/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
  - [hacktoberfest](https://github.com/Azure/azure-dev/issues?q=is%3Aopen+is%3Aissue+label%3Ahacktoberfest)

**Reply by himanshu1221** [reply]
Hey @lechnerc77 thanks for answering my question!
Can you Guide me on how can I spin up a devcontainer in codespace?

**Reply by lechnerc77** [reply]
I think the official documentation is a good starzing point in both cases:

- [Development Containers](https://code.visualstudio.com/docs/devcontainers/containers) - VSCode specific
- [GitHub Codespaces](https://docs.github.com/en/codespaces/overview)
